framework/src/config/config_reader.rs

```diff
@@ -13,297 +14,109 @@ pub const DEFAULT_NAME: &'static str = "zcsi";
-            return Err(ErrorKind::ConfigurationError(String::from("Could not parse name")).into());
-        }
-    };
+#[derive(Debug, Deserialize)]
+pub struct RemoteConfiguration {
+    pub host: String,
+    pub port: u16,
+    pub healthy: bool,
+}
 
-    // Get primary core from configuration.
-    let master_lcore = match toml.get("master_core") {
-        Some(&Value::Integer(core)) => core as i32,
-        Some(&Value::String(ref core)) => match core.parse() {
-            Ok(c) => c,
-            _ => {
-                return Err(ErrorKind::ConfigurationError(format!(
-                    "Could not parse {} as core",
-                    core
-                )).into())
-            }
-        },
-        None => DEFAULT_PRIMARY_CORE,
-        v => {
-            println!("Could not parse core");
-            return Err(
-                ErrorKind::ConfigurationError(format!("Could not parse {:?} as core", v)).into(),
-            );
+impl Default for RemoteConfiguration {
+    fn default() -> RemoteConfiguration {
+        RemoteConfiguration {
+            host: String::from("::1"),
+            port: 1285,
+            healthy: false,
         }
-    };
+    }
+}
 
-    // Parse mempool size
-    let pool_size = match toml.get("pool_size") {
-        Some(&Value::Integer(pool)) => pool as u32,
-        None => DEFAULT_POOL_SIZE,
-        _ => {
-            println!("Could parse pool size");
-            return Err(
-                ErrorKind::ConfigurationError(String::from("Could not parse pool size")).into(),
-            );
-        }
-    };
+#[derive(Debug, Deserialize)]
+pub struct AgentConfiguration {
+    pub host: String,
+    pub port: u16,
+    pub healthy: bool,
+}
 
-    // Get cache size
-    let cache_size = match toml.get("cache_size") {
-        Some(&Value::Integer(cache)) => cache as u32,
-        None => DEFAULT_CACHE_SIZE,
-        _ => {
-            println!("Could parse cache size");
-            return Err(
-                ErrorKind::ConfigurationError(String::from("Could not parse cache size")).into(),
-            );
+impl Default for AgentConfiguration {
+    fn default() -> AgentConfiguration {
+        AgentConfiguration {
+            host: String::from("::1"),
+            port: 1347,
+            healthy: false,
         }
-    };
+    }
+}
 
-    // Is process a secondary process
-    let secondary = match toml.get("secondary") {
-        Some(&Value::Boolean(secondary)) => secondary,
-        None => DEFAULT_SECONDARY,
-        _ => {
-            println!("Could not parse whether this is a secondary process");
-            return Err(ErrorKind::ConfigurationError(String::from(
-                "Could not parse secondary processor spec",
-            )).into());
-        }
-    };
+/// Read a configuration file and create a `NetbricksConfiguration` structure.
+/// `filename` should be TOML formatted file.
+pub fn read_configuration_file(filename: &str) -> Result<OccamConfiguration> {
```

Comment:
in Rust, do we explicitly define the result failure type or nah?

Reply:
On the `Result` enum it does require it, but I think we got a type aliased through the imports somewhere.

`type Result<T> = Result<T, SomeErrorType>;`

I'm just too lazy to track down where it's coming from. I think it's from `io`.